core/src/services/webdav/backend.rs

```diff
@@ -471,6 +472,79 @@ impl WebdavBackend {
         self.client.send(req).await
     }
 
+    // read will send http request for dav access.
+    // sometimes we need to send request to different endpoint when server side
+    // indicate we need to redirect to another url.
+    #[async_recursion]
```

Comment:
We don't need to do recursion, we can expand the redirection by hand. And we should add a new config value for the redirection limit (maybe default to 3 times).

Reply:
Hi i have modified it according to the review.

I also did some changes to pass all CI tests. Please help have one look on this if you got time